open_feature/open_feature_client.py

```diff
@@ -250,15 +251,11 @@ def evaluate_flag_details(
         # in the flag evaluation
         # before: API, Client, Invocation, Provider
         merged_hooks = (
-            self.hooks
-            + flag_evaluation_options.hooks
-            + self.provider.get_provider_hooks()
+            self.hooks + evaluation_hooks + self.provider.get_provider_hooks()
         )
         # after, error, finally: Provider, Invocation, Client, API
         reversed_merged_hooks = (
-            self.provider.get_provider_hooks()
-            + flag_evaluation_options.hooks
-            + self.hooks
+            self.provider.get_provider_hooks() + evaluation_hooks + self.hooks
```

Comment:
I'd recommend just setting the reverse of the other one, such that it reduces the likelihood we update it in one place and not the other.

Reply:
This is done now the same way as the other SDKs.

This would technically result in different hook execution behavior if any of the hook lists has more than one entry. As it is listed in [the spec](https://docs.openfeature.dev/docs/specification/sections/hooks#requirement-442):

> Hooks **MUST** be evaluated in the following order:
>
> - before: API, Client, Invocation, Provider
> - after: Provider, Invocation, Client, API
> - error (if applicable): Provider, Invocation, Client, API
> - finally: Provider, Invocation, Client, API

The terminology in the specification does not appear to be strict enough to dictate this behavior in one such manner, such that both this Python implementation and the JavaScript one technically appease the spec but currently result in different hook behaviors in the case that any of the hook lists contain more than one hook.

I opened [an issue in the specification](https://github.com/open-feature/spec/issues/164#issue-1465298095) in order to bring clarity around the behavior of the latter 3 hooks.

Reply:
I think regardless of how this spec issue proceeds, we can move forward with this PR. If we do make a spec change, the normal process is to review all SDKs, so it will be addressed if/when that happens.

Reply:
Yep, I also considered this when I reviewed the original implementation. As @tcarrio says, the current implementation ensures that each hook type is implemented in an order defined by the user, within the global ordering of the different hook types. 

I agree that we should resolve this conversation and update it if / when the spec is updated.